packages/cactus-plugin-satp-hermes/src/main/typescript/gol/crash-manager.ts

```diff
@@ -0,0 +1,726 @@
+
+    const satpLoggerConfig: ISATPLoggerConfig = {
+      localRepository: this.localRepository,
+      remoteRepository: this.remoteRepository,
+      signer: this.signer,
+      pubKey: this._pubKey,
+    };
+
+    this.dbLogger = new SATPLogger(satpLoggerConfig);
+    this.log.debug(`${fnTag} dbLogger initialized: ${!!this.dbLogger}`);
+
+    this.factory = new RollbackStrategyFactory(
+      this.bridgesManager,
+      this.log,
+      this.dbLogger,
+    );
+
+    this.crashRecoveryServerService = new CrashRecoveryServerService(
+      this.bridgesManager,
+      this.localRepository,
+      this.sessions,
+      this.dbLogger,
+      this.signer,
+      this.log,
+    );
+
+    this.crashRecoveryClientService = new CrashRecoveryClientService(
+      this.dbLogger,
+      this.log,
+      this.signer,
+    );
+
+    this.crashRecoveryHandler = new CrashRecoveryHandler(
+      this.crashRecoveryServerService,
+      this.crashRecoveryClientService,
+      this.log,
+    );
+
+    const crashRecoveryHandlers = new Map<string, SATPHandler>();
+    crashRecoveryHandlers.set("crash-handler", this.crashRecoveryHandler);
+    this.orchestrator.addHandlers(crashRecoveryHandlers);
+  }
+
+  get className(): string {
+    return CrashManager.CLASS_NAME;
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public async recoverSessions() {
+    const fnTag = `${this.className}#recoverSessions()`;
+
+    try {
+      const allLogs = await this.localRepository.readLogsNotProofs();
+      for (const log of allLogs) {
+        const sessionId = log.sessionId;
+        this.log.info(
+          `${fnTag}, recovering session from database: ${sessionId}`,
+        );
+
+        if (!log || !log.data) {
+          throw new Error(`${fnTag}, invalid log`);
+        }
+
+        try {
+          const sessionData: SessionData = JSON.parse(log.data);
+          const satpSession = SATPSession.getSession(sessionData);
+          this.sessions.set(sessionId, satpSession);
+        } catch (error) {
+          this.log.error(
+            `Error parsing log data for session Id: ${sessionId}: ${error}`,
+          );
+        }
+      }
+
+      if (this.sessions.size === 0) {
+        this.log.info(`${fnTag} No active sessions!`);
+      } else {
+        this.crashScheduler = schedule.scheduleJob(
+          this.options.healthCheckInterval || "*/5 * * * * *",
+          async () => {
+            await this.checkAndResolveCrashes();
+          },
+        );
+        this.log.info(`${fnTag} Crash detection job is running`);
+      }
+    } catch (error) {
+      this.log.error(`Error initializing sessions: ${error}`);
+    }
+  }
+
+  private updateSessionState(sessionId: string, newState: State): string {
+    const session = this.sessions.get(sessionId);
+    if (!session) {
+      throw new Error(`Session with ID ${sessionId} not found.`);
+    }
+
+    const sessionData = session.hasClientSessionData()
```

Comment:
This needs to be resolved!!
This is a issue when the gateway is either the client and the server.